### ISSUE TITLE
Custom Theme Not Working with @mui/material 6.0.0 and pigment-css in Next.js

### ISSUE THREAD

**Issue by wh5938316** [mui/pigment-css#215]
### Steps to reproduce

next.config.mjs
```
export default withPigment(nextConfig, {
  theme: createTheme({
    cssVariables: true,
    colorSchemes: { light: true, dark: true },
    typography: {
      fontFamily: 'var(--font-roboto)',
    },
    components: {
      MuiButton: {
        styleOverrides: {
          root: {
            background: '#000 !important',
          },
        },
      }
    }
  }),
  transformLibraries: ['@mui/material'],
});
```

The custom theme is not being applied as expected. Components do not reflect the customized theme settings.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: pigment-css, next.js

**Comment by wh5938316** [comment]
The issue has been taken over by the @mui/material maintainers.
[@mui/material issue #43501](https://github.com/mui/material-ui/issues/43501)

**Comment by wh5938316** [comment]
Resolved after 0.0.21 release
